src/collective/behavior/featuredimage/subscriber.py

```diff
@@ -15,9 +19,16 @@ def _get_screenshot(page):
     driver.get(page)
     data = driver.get_screenshot_as_png()
     # crop image
-    el = driver.find_element_by_id('featuredimage')
-    location = el.location
-    size = el.size
+    try:
+        el = WebDriverWait(driver, 10).until(
+            EC.presence_of_element_located((By.ID, 'featuredimage'))
+        )
+        # Put these lines after driver.quit will raise URLError exception
+        location = el.location
```

Comment:
if the previous line fails then we will have an exception below this block; we need to review this.